### DISCUSSION TITLE
OpenGL backend

### DISCUSSION THREAD

**Discussion by wangjiawen2013** [tracel-ai/burn#2313] [Q&A] [answered]
Hi,
Does burn support OpenGL backend on linux ?  Could you give an example ?

**Comment by AsherJingkongChen** [comment, marked as answer]
Via wgpu since it has GL backend: https://wgpu.rs

**Reply by laggui** [reply]
Exactly 🙂 

You're using it properly [here](https://github.com/tracel-ai/burn/issues/1691#issuecomment-2375555989), just looks like you don't have a valid setup so wgpu is not able to find an adapter.
